File: app/assets/javascripts/models/measure.js


Cibi.Measure = DS.Model.extend({
	fieldName: DS.attr('string'),
	formatAs: DS.attr('string'),
	displayName: DS.attr('string'),
	sortOrder: DS.attr('string'),
	isCalculated: DS.attr('boolean'),
	prefix: DS.attr('string'),
	suffix: DS.attr('string'),
	unit: DS.attr('string'),

	chart: DS.belongsTo('Cibi.Chart'),
	accountTemplate: DS.belongsTo('Cibi.AccountTemplate'),

	factUnit:function(){
		unit = {};
		unit['prefix'] = this.get('prefix');
		unit['suffix'] = this.get('suffix');
		return unit;
	}.property('prefix', 'suffix'),
});

File: app/models/measure.rb
class Measure < AuditedModel
  attr_accessible :chart_id, :display_name, :field_name, :format_as, :is_calculated, :prefix, 
  					:sort_order, :suffix, :unit, :account_template_id

  belongs_to :chart
  belongs_to :account_template
  validates :field_name, :presence => true
  
  
  	def get_fact_map_entry
		return {
			"fieldName" => self.field_name,
			"formatAs" => self.format_as,
			"displayName" => self.display_name,
		}
  	end 
end
